File: app/src/main/java/com/example/bookworm/Search/items/Book.java
package com.example.bookworm.Search.items;

import java.io.Serializable;

public class Book implements Serializable {
    //제목, 간단한 내용, 출판사, 저자 정도 표기?
    private String img_url;
    private String title;
    private String content;
    private String publisher;
    private String author;
    private String itemId;


    public Book(String... strings) {

        this.title = strings[0];
        this.content = strings[1];
        this.publisher = strings[2];
        this.author = strings[3];
        this.img_url = strings[4];
        if (strings.length == 6) this.itemId = strings[5];

    }


    public String getImg_url() {
        return img_url;
    }

    public String getItemId() {
        return itemId;
    }

    public String getPublisher() {
        return publisher;
    }

    public String getAuthor() {
        return author;
    }

    public String getTitle() {
        return title;
    }

    public String getContent() {
        return content;
    }

}

File: app/src/main/java/com/example/bookworm/bottomMenu/search/bookitems/Book.kt
package com.example.bookworm.bottomMenu.search.bookitems

import android.os.Parcelable
import com.google.firebase.firestore.Exclude
import kotlinx.parcelize.Parcelize

//도서 DTO(Data Transfer Object)
@Parcelize
data class Book(
        var imgUrl: String = "", //도서 이미지
        var title: String = "",
        var categoryName: String = "",
        var content: String = "",
        var publisher: String = "",
        var author: String = "",
        var itemId: String = "",
        var isbn: String = "",
        @Exclude @get:Exclude @set:Exclude
        var originPrice:String ="", //기존 가격
        @Exclude @get:Exclude @set:Exclude
        var salePrice:String = "", //판매 가격
         @Exclude @get:Exclude @set:Exclude
        var userRate: Float = 0F, //평점 점수
        @Exclude @get:Exclude @set:Exclude
        var isRecommend: Boolean = false,
        @Exclude @get:Exclude @set:Exclude
        var purchaseLink:String ="" //구매링크
) : Parcelable